File: Java/1673. Find the Most Competitive Subsequence.java
class Solution {
    public int[] mostCompetitive(int[] nums, int k) {
        int n = nums.length;
        int j = 0;
        int[] res = new int[k];
        for(int i = 0; i < n; i++){
            while(j != 0 && n - i > k - j && res[j - 1] > nums[i]){
                j--;
            }
            if(j < k){
                res[j] = nums[i];
                j++;
            }
        }
        return res;
    }
}

File: JavaScript/1673. Find the Most Competitive Subsequence.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var mostCompetitive = function(nums, k) {
    var n = nums.length;
    var j = 0;
    var res = new Array(k).fill(0);
    for(var i = 0; i < n; i++){
        while(j != 0 && n - i > k - j && res[j - 1] > nums[i]){
            j--;
        }
        if(j < k){
            res[j] = nums[i];
            j++;
        }
    }
    return res;
};
